src/bitcoin/checkpoint.rs

```diff
@@ -1204,21 +1223,20 @@ impl<'a> BuildingCheckpointMut<'a> {
 
             // Find the first remaining output of the intermediate tx which
             // matches the amount being spent by this final tx's input.
-            for (i, output) in intermediate_tx_outputs.iter() {
+            for (i, (vout, output)) in intermediate_tx_outputs.iter().enumerate() {
                 if output == &(input.amount) {
                     // Once found, link the final tx's input to the vout index
                     // of the the matching output from the intermediate tx, and
                     // remove it from the matching list.
 
                     input.prevout = Adapter::new(bitcoin::OutPoint {
                         txid: intermediate_tx_id,
-                        vout: *i as u32,
+                        vout: *vout as u32,
                     });
-                    intermediate_tx_outputs.remove(*i);
-
                     // Deduct the final tx's miner fee from its outputs,
                     // removing any outputs which are too small to pay their
                     // share of the fee.
+                    intermediate_tx_outputs.remove(i);
```

Comment:
Can we move this line up so that it's not under this comment? It is misleading to be here since it is not relevant to them comment.